src/woo_publications/publications/api/filters.py

```diff
@@ -2,7 +2,32 @@
 
 from django_filters.rest_framework import FilterSet, filters
 
-from ..models import Publication
+from ..models import Document, Publication
+
+
+class DocumentFilterSet(FilterSet):
+    publicatie = filters.UUIDFilter(
+        field_name="publicatie__uuid",
+        lookup_expr="exact",
+        help_text=_(
+            "Search the document based on the unique identifier that represents a publication."
+        ),
+    )
```

Comment:
You can make this more semantic by using the filters for relations:

```suggestion
@extend_schema_field(UUID)
class UUIDModelChoiceFilter(filters.ModelChoiceFilter):
    pass


class DocumentFilterSet(FilterSet):
    publicatie = UUIDModelChoiceFilter(
        queryset=Publication.objects.all(),
        to_field_name="uuid",
        help_text=_(
            "Search the document based on the unique identifier that represents a publication."
        ),
    )
```

It requires some drf-annotation gymnastics, but looks like you can get there relatively easy.

Not sure why the help text is lost though